hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/Transformer.java

```diff
@@ -45,4 +47,11 @@ public interface Transformer {
    */
   @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
   Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties);
+
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  default Option<StructType> transformedSchema(JavaSparkContext jsc, SparkSession sparkSession, StructType incomingStruct, TypedProperties properties) {
```

Comment:
Won't we always have a schema returned? Do we need Option here?

Reply:
Addressed